From 7c978096fd4ad0335387be8c23078b80261a01d4 Mon Sep 17 00:00:00 2001
From: Hendrik Leppkes <h.leppkes@gmail.com>
Date: Mon, 12 Mar 2012 22:04:19 +0100
Subject: [PATCH 07/65] Revert "mov: Set negative Sample_duration in STTS to 1"

This reverts commit 1892052f8dee858dcab3292d1c40c5deef67e18f.
Some files may violate the spec, but this way they play properly.
---
 libavformat/mov.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/libavformat/mov.c b/libavformat/mov.c
index 64e2824..c4ed03d 100644
--- a/libavformat/mov.c
+++ b/libavformat/mov.c
@@ -2206,12 +2206,6 @@ static int mov_read_stts(MOVContext *c, AVIOContext *pb, MOVAtom atom)
         sample_count=avio_rb32(pb);
         sample_duration = avio_rb32(pb);
 
-        /* sample_duration < 0 is invalid based on the spec */
-        if (sample_duration < 0) {
-            av_log(c->fc, AV_LOG_ERROR, "Invalid SampleDelta %d in STTS, at %d st:%d\n",
-                   sample_duration, i, c->fc->nb_streams-1);
-            sample_duration = 1;
-        }
         if (sample_count < 0) {
             av_log(c->fc, AV_LOG_ERROR, "Invalid sample_count=%d\n", sample_count);
             return AVERROR_INVALIDDATA;
-- 
2.3.5

